v2/pkg/engine/resolve/loader.go

```diff
@@ -24,6 +25,17 @@ import (
 	"github.com/wundergraph/graphql-go-tools/v2/pkg/pool"
 )
 
+type LoaderHooks interface {
+	// OnLoad is called before the fetch is executed
+	OnLoad(ctx context.Context, dataSourceID string) context.Context
+	// OnFinished is called after the fetch has been executed and the response has been processed and merged
+	OnFinished(ctx context.Context, statusCode int, dataSourceID string, err error)
+}
+
+func IsIntrospectionDataSource(dataSourceID string) bool {
+	return dataSourceID == "introspection__schema&__type" || dataSourceID == "introspection__type__fields" || dataSourceID == "introspection__type__enumValues"
```

Comment:
could we move these string values into constants in this package and reuse them in the introspection datasource?

Reply:
My plan was to put this in the same package where we defined the datasources but I run into circular dependencies. I can export them as const in this pkg.

Reply:
Updated, please check.

Reply:
Yeah, I know about circulars, that is why I proposed to put it here

The missing part is to reuse these consts in datasource as IDs

Reply:
Done